server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapper.java

```diff
@@ -115,8 +123,13 @@ public Query termQuery(Object value, SearchExecutionContext context) {
         }
     }
 
-    private TimeSeriesIdFieldMapper() {
+    private final IndexVersion indexCreatedVersion;
+    private final boolean useDocValuesSkipper;
```

Comment:
I think we can avoid passing down index version here. We can set `useDocValuesSkipper` based on the following expressions: `indexVersion.onOrAfter(IndexVersions.TIME_SERIES_ID_DOC_VALUES_SPARSE_INDEX) && context.getIndexSettings().useDocValuesSkipper()`?